discovery_engine_core/web-api/openapi.yaml

```diff
@@ -0,0 +1,160 @@
+        specified `user_id`. The score is a value between 0 and 1 where a higher value
+        means that the document matches the preferences of the user better.
+      parameters:
+        - name: user_id
+          in: path
+          description: Id of the user
+          required: true
+          schema:
+            $ref: '#/components/schemas/UserId'
+        - name: count
+          in: query
+          description: Maximum number of personalized documents to return
+          type: integer
+          format: int32
+          minimum: 1
+          maximum: 100
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/PersonalizedDocuments'
+        '400':
+          description: invalid user id
+        '404':
+          description: user not found or the user does not have enough interactions
+
+  /users/{user_id}/documents/interacted:
+    post:
+      tags:
+        - personalization
+      summary: Add interaction between a user and a document
+      description: |-
+        The interaction is used to provide personalized documents to the user.
+      parameters:
+        - name: user_id
+          in: path
+          description: Id of the user
+          required: true
+          schema:
+            $ref: '#/components/schemas/UserId'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/UserInteraction'
+      responses:
+        '204':
+          description: successful operation
+        '400':
+          description: invalid request. User or document id is invalid
+        '404':
+          description: user not found
+        
+  /documents/:
+    post:
+      tags:
+        - ingestion
+      summary: Add documents to the system
+      description: |-
+        Add documents to the system. The system will create a representation of the document
+        that will be used to match it against the preferences of a user.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/IngestionRequest'
+      responses:
+        '200':
+          description: successful operation
+        '400':
+          description: invalid request
+        '422':
+          description: all or some of the documents were not successfully uploaded
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/UserInteraction'
+  
+components:
+  schemas:
+    DocumentId:
+      type: string
+      example: "document_id_1"
+    UserId:
+      type: string
+      example: "user_id_1"
+    PersonalizedDocumentData:
+      type: object
+      properties:
+        score:
+          type: number
+      example:
+        - score: 0.872
+    PersonalizedDocuments:
+      type: object
```

Comment:
souldn't that be an array where the items are an object with an id + the data
also the example might be updated because currently it is `example: List [ OrderedMap { "document_id_1": List [ OrderedMap { "score": 0.872 } ] } ]`

Reply:
I'm not sure why the fact that is a response should imply that is an object, but is a good idea. I'll go with
```js
{
  documents: [
    {
       id: "document_id",
       score: 0.1
     }
  ]
}
```
The list is not necessary ordered by score.

Reply:
There are multiple reasons:

1. legacy reason: various security guidelines requires json response bodies to be objects due to some security vulnerability wrt. JS and old browsers. This doesn't apply anymore.
2. convention, due to 1. and 3. it had become a de-facto convention for many developers
3. and the main reason: extend-ability. We can always add fields to an object if we need to return additional information in the future making it a non-braking change. We can't do so with an array. This makes the "always return objects" approach much less prone to accidental braking changes.